File: 6-kyu/1rm-calculator/python/solution.py
def epley(w, r):
    return w * (1 + r / 30)

def mcglothin(w, r):
    return 100 * w / (101.3 - 2.67123 * r)

def lombardi(w, r):
    return w * r ** 0.1

def calculate_1RM(w, r):
    if r < 1:
        return 0
    elif r == 1:
        return w
    return max(round(epley(w, r)), round(mcglothin(w, r)), round(lombardi(w, r)))

File: 6-kyu/1rm-calculator/swift/solution.swift
import Foundation

func calculate1RM(_ weight: Int, _ reps: Int) -> Int? {
    if reps == 0 {
        return nil
    }
    else if reps == 1 {
        return weight
    }
    let epley = Double(weight) * (1 + Double(reps) / 30)
    let mcglothin = 100 * Double(weight) / (101.3 - 2.67123 * Double(reps))
    let lombardi = Double(weight) * pow(Double(reps), 0.1)
    let results: [Double] = [epley.rounded(), mcglothin.rounded(), lombardi.rounded()]
    return Int(results.max()!)
}